C++/ShiftSort.cpp

```diff
@@ -117,7 +117,7 @@ void ShiftSort::merge(int *array, int first_index, int second_index, int third_i
       while(second_counter>0)
       {
           //shift if left is greater than right
-          if(left>=first_index && array[left]>=temp_2nd[second_counter-1])
+          if(left>=first_index && array[left]>temp_2nd[second_counter-1])
```

Comment:
Must be > and not >=. The reason is that the merging is done from back to front. If we have an equal element, then the element from the temporary vector should be taken to keep correct order.